### PR TITLE
added `safety check` to python dependencies

### LINKED ISSUE #2451: Add `safety check` for all python dependencies to CI
# Feature request

As noted in #2450, we have a vulnerable dependency. We should automate checks for this.

## Description

In our travis build:

```bash
pip install safety
find . -name '*requirements.txt' -print0 | xargs -0 -n 1 safety check --full-report -r
```

## User Stories

As a dev, I don't want to manually check for vulns in my dependencies.

As an admin, I don't want to get hacked via my dependencies.

As a server, I don't want to get hacked at all kthx.

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for review

## Description of Changes

Fixes #2451

Adds `safety check` to a few scripts and `Makefile`s as well as CI.

## Testing

`make safety`

This breaks right now as we have a CVE in the code. Once it's removed, we can merge this in.

### CODE REVIEW

**Comment by dachary** [comment]
I confirm circleci fails as expected https://circleci.com/gh/freedomofpress/securedrop/2960?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
<pre>
-> ansible, installed 2.2.1, affected <2.3.1, id 34941
ansible before 2.3.1 is vulnerable to CVE-2017-7481 - data for lookup plugins used as variables was not being correctly marked as "unsafe".
--
</pre>

**Comment by dachary** [comment]
I marked as **PR: pending additional work** because it needs to wait on the CVE to be dealt with. I did not know about this way to find CVE, interesting !

File: testinfra/requirements.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #
 #    pip-compile --output-file requirements.txt requirements.in
 #
-ansible-lint==3.4.13      # via molecule
+ansible-lint==3.4.16      # via molecule
```
Comment (dachary) [inline testinfra/requirements.txt]: In the spirit of keeping this PR to the minimum, would you mind stripping out the module upgrades that are unrelated to the `safety` module ?

Comment (heartsucker) [reply testinfra/requirements.txt]: I could, but I was using `pip-compile` and I don't actually know if there's any weird breakages, so I'm letting that tool do it's magic. This is something one of the ops homies would need to comment on.

Comment (dachary) [reply testinfra/requirements.txt]: FWIW when I did the same in the past I just manually removed all dependencies that were not strictly related to the module I was adding. I don't think there is a way to ask pip-compile to focus on a single module and leave the others alone.

Comment (msheiny) [reply testinfra/requirements.txt]: Yeah @dachary is correct here ;) Unfortunately the pip-compile story sucks for just updating one dependency. You could temporarily peg all the dependencies in the `requirements.in` file and then run it ... that's a hacky way around it. Or yeah, do what @dachary said and try to manually edit the commit changes using something like `git add -p`

**Comment by redshiftzero** [comment]
Once #2483 is merged in you can rebase on this and tests should pass :sparkles:

**Comment by heartsucker** [comment]
Pinging for merge now that CI his happy.

**Comment by dachary** [comment]
CircleCI is happy

<pre>
#!/bin/bash -eo pipefail
make safety

Checking file ./install_files/ansible-base/requirements.txt
safety report
checked 15 packages, using default DB
---
No known security vulnerabilities found.


Checking file ./testinfra/requirements.txt
safety report
checked 72 packages, using default DB
---
No known security vulnerabilities found.


Checking file ./securedrop/requirements/develop-requirements.txt
safety report
checked 61 packages, using default DB
---
No known security vulnerabilities found.


Checking file ./securedrop/requirements/test-requirements.txt
safety report
checked 28 packages, using default DB
---
No known security vulnerabilities found.


Checking file ./securedrop/requirements/securedrop-requirements.txt
safety report
checked 25 packages, using default DB
---
No known security vulnerabilities found.
</pre>

**Pull request merged**